src/test/java/io/rtr/conduit/amqp/impl/AMQPTransportTest.java

```diff
@@ -25,7 +33,7 @@
 
     @Before
     public void before() {
-        amqpTransport = spy(new AMQPTransport("host", 1234));
```

Comment:
can you mock the ConnectionFactory class and see that the method `setSocketFactory` is called?

Reply:
I can, just requires moving stuff around, creating fields to hold these mocked values.  Honestly, I don't think checking whether I called `setSocketFactory` with SSLSocketFactory is that useful of a test, it's like writing a test that tests that I wrote specific code, I'd rather test a behavior, e.g. that I can now connect to SSL, however it's achieved.  That's why I've been on the fence about how to test this.  I did write an integration test to test this personally, but I didn't keep it since it had IBM cloud credentials.

Reply:
yeah, i agree with it not testing behavior if you go that route